From 0658dc4e4a5961eaa846d0801a03cbc3acd84138 Mon Sep 17 00:00:00 2001
From: Victor Kaplansky <victork@redhat.com>
Date: Fri, 4 Mar 2016 10:45:31 +0100
Subject: [PATCH 1/3] Revert "vhost: send SET_VRING_ENABLE at start/stop

RH-Author: Victor Kaplansky <victork@redhat.com>
Message-id: <1457088132-9937-1-git-send-email-victork@redhat.com>
Patchwork-id: 69546
O-Subject: [RHEV-7.3/7.2.z qemu-kvm-rhev PATCH] Revert "vhost: send SET_VRING_ENABLE at start/stop
Bugzilla: 1316156
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>

From: "Michael S. Tsirkin" <mst@redhat.com>

This reverts commit 3a12f32229a046f4d4ab0a3a52fb01d2d5a1ab76.

In case of live migration several queues can be enabled and not only the
first one. So informing backend that only the first queue is enabled is
wrong.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1310543
Brew build: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=10599692

Note: lightly tested in my laptop with Fedora 22.

Reported-by: Thibaut Collet <thibaut.collet@6wind.com>
Cc: Yuanhan Liu <yuanhan.liu@linux.intel.com>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Reviewed-by: Yuanhan Liu <yuanhan.liu@linux.intel.com>
(cherry-picking commit 449e3578)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/virtio/vhost.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/hw/virtio/vhost.c b/hw/virtio/vhost.c
index a04db26..4e5f7cd 100644
--- a/hw/virtio/vhost.c
+++ b/hw/virtio/vhost.c
@@ -1210,11 +1210,6 @@ int vhost_dev_start(struct vhost_dev *hdev, VirtIODevice *vdev)
         }
     }
 
-    if (hdev->vhost_ops->vhost_backend_set_vring_enable) {
-        /* only enable first vq pair by default */
-        hdev->vhost_ops->vhost_backend_set_vring_enable(hdev, hdev->vq_index == 0);
-    }
-
     return 0;
 fail_log:
     if (hdev->log_size) {
@@ -1247,10 +1242,6 @@ void vhost_dev_stop(struct vhost_dev *hdev, VirtIODevice *vdev)
                              hdev->vq_index + i);
     }
 
-    if (hdev->vhost_ops->vhost_backend_set_vring_enable) {
-        hdev->vhost_ops->vhost_backend_set_vring_enable(hdev, 0);
-    }
-
     vhost_log_put(hdev, true);
     hdev->started = false;
     hdev->log = NULL;
-- 
1.8.3.1

